### PR TITLE
equilibrium index in an array

### LINKED ISSUE #9855: Find Equilibrium index in an array
### Feature description

The equilibrium index of an array is an index such that the sum of elements at lower indexes is equal to the sum of elements at higher indexes.

### PR DESCRIPTION
**Pull request by Arunsiva003**
### Describe your change:

fixes #9855
closing #9855

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/arrays/equilibrium_index_in_array.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+"""
+
+
+def equilibrium_index(arr: list[int], size: int) -> int:
+    """
+    Find the equilibrium index of an array.
+
+    Args:
+        arr (List[int]): The input array of integers.
+        n (int): The size of the array.
```
Comment (ChrisO345) [inline data_structures/arrays/equilibrium_index_in_array.py:29]: Please do not reference types within docstrings

Comment (Arunsiva003) [reply data_structures/arrays/equilibrium_index_in_array.py]: done

Comment (Arunsiva003) [reply data_structures/arrays/equilibrium_index_in_array.py]: Review needed!!

Comment (Arunsiva003) [reply data_structures/arrays/equilibrium_index_in_array.py]: > Please do not reference types within docstrings

Done!!
Review need to merge

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**